### PR TITLE
TIAF Python Opt-In: Phase 1

### PR DESCRIPTION
**Pull request by jonawals**
## What does this PR do?

This PR enables the TIAF Python stage and opts-in the the White Box Python tests.

## How was this PR tested?

Manual testing of AR.

### CODE REVIEW

File: scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py [hunk 1]
```diff
@@ -42,7 +42,10 @@ def runtime_type(self):
 
 class TestTiafDriver():
 
-    def test_run_Tiaf_mars_index_prefix_is_supplied(self, caplog, main_args, mock_runtime, mocker):
+    def test_run_Tiaf_mars_index_prefix_is_supplied(self, caplog, main_args, test_targets_enabled, mock_runtime, mocker):
+        if not test_targets_enabled:
```
Comment (evanchia-ly-sdets) [inline scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py:46]: If all tests in the class have the same requirement, next time you can use a setUp() at the class level for all shared setup steps.

**Review by evanchia-ly-sdets — APPROVED** [review]

**Review by brianherrera — APPROVED** [review]

File: scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py [hunk 2]
```diff
@@ -42,7 +42,10 @@ def runtime_type(self):
...
+            return
```
Comment (Kadino) [inline scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py:47]: Seems more appropriate to use [pytest.mark.skipif()](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=skipif#pytest.mark.skipif) or an inline `pytest.skip()` to indicate when these tests have nothing to run. A return will mark a test as a success, which tends to be misleading if none of its asserts are executed.

**Review by Kadino — APPROVED** [review]
No concerns for enabling conditional skipping of WhiteBox tests in a branch, though you will likely need buy in from SIG-Content when merging to development.

**Review by amzn-changml — APPROVED** [review]

**Review by smurly — APPROVED** [review]

File: scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py (hunk 1 above)
Comment (jonawals) [reply scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py]: Fixed 👍

**Pull request merged**